File: app/src/main/java/com/konektedi/vs/Home/Candidates/CandidatesRepository.java
package com.konektedi.vs.Home.Candidates;


import android.arch.lifecycle.MutableLiveData;

import com.konektedi.vs.R;
import com.konektedi.vs.Utilities.Api.ApiUtilities;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

import static com.konektedi.vs.Home.Candidates.Candidates.getContextOfCandidates;

public class CandidatesRepository {

    public MutableLiveData<List<CandidatesModel>> getCandidates(String election_id, String category_id) {

        ((Candidates) getContextOfCandidates()).showProgressBar();

        final MutableLiveData<List<CandidatesModel>> listMutableLiveData = new MutableLiveData<>();

        Call<List<CandidatesModel>> call = ApiUtilities.getClient().getCandidates(election_id, category_id);

        call.enqueue(new Callback<List<CandidatesModel>>() {
            @Override
            public void onResponse(Call<List<CandidatesModel>> call, Response<List<CandidatesModel>> response) {
                ((Candidates) getContextOfCandidates()).hideProgressBar();

                if (response.isSuccessful()) {
                    List<CandidatesModel> categoriesModelList = response.body();
                    listMutableLiveData.setValue(categoriesModelList);

                } else if (response.code() == 404) {
                    ((Candidates) getContextOfCandidates()).showAlert(R.string.no_candidates);
                }
            }

            @Override
            public void onFailure(Call<List<CandidatesModel>> call, Throwable t) {
                ((Candidates) getContextOfCandidates()).hideProgressBar();
                ((Candidates) getContextOfCandidates()).showAlert(R.string.error);
            }
        });

        return listMutableLiveData;
    }
}

File: app/src/main/java/com/konektedi/vs/home/candidates/CandidatesRepository.kt
package com.konektedi.vs.home.candidates


import android.arch.lifecycle.MutableLiveData
import android.util.Log

import com.konektedi.vs.utilities.NetworkStatus
import com.konektedi.vs.utilities.api.ApiN
import com.konektedi.vs.utilities.api.ApiUtilities
import com.konektedi.vs.utilities.api.getError
import com.konektedi.vs.utilities.common.NetworkState
import com.konektedi.vs.utilities.models.Candidate
import okhttp3.ResponseBody

import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class CandidatesRepository {
    private val apiClient = ApiN.create()
    val mNetworkState = MutableLiveData<NetworkState>()

    fun getCandidates(electionId: Int, categoryId: Int): MutableLiveData<List<Candidate>> {
        mNetworkState.value = NetworkState.LOADING
        val categoriesList = MutableLiveData<List<Candidate>>()

        apiClient.getCandidates(electionId, categoryId).enqueue(
                object : Callback<List<Candidate>> {
                    override fun onFailure(call: Call<List<Candidate>>, t: Throwable) {
                        mNetworkState.value = NetworkState.serverMsg(t.message) //TODO Weka failed to connect basi.
                    }

                    override fun onResponse(call: Call<List<Candidate>>, response: Response<List<Candidate>>) {
                        when {
                            response.isSuccessful -> {
                                mNetworkState.value = NetworkState.LOADED
                                categoriesList.value = response.body()
                            }
                            else -> {
                                mNetworkState.value = NetworkState.serverMsg(getError(response as Response<ResponseBody>))
                            }
                        }
                    }
                })
        return categoriesList
    }

    //TODO check
//    fun checkVote(electionId: Int, categoryId: Int): MutableLiveData<List<Candidate>> {
//        mNetworkState.value = NetworkState.LOADING
//
//        apiClient.getCandidates(electionId, categoryId).enqueue(
//                object : Callback<List<Candidate>> {
//                    override fun onFailure(call: Call<List<Candidate>>, t: Throwable) {
//                    }
//
//                    override fun onResponse(call: Call<List<Candidate>>, response: Response<List<Candidate>>) {
//
//                    }
//                })
//    }
}

